corehq/apps/data_interfaces/views.py

```diff
@@ -511,6 +519,38 @@ def post(self, request, *args, **kwargs):
             )
         )
 
+    def inaccessible_forms_accessed(self, xform_ids, domain, couch_user):
```

Comment:
This should have something like this at the top:
```python
if couch_user.has_permission(self.domain, 'access_all_locations'):
    return True
```
so we don't needlessly run this logic (and that query) on non-restricted users.